package/MDAnalysis/analysis/waterdynamics.py

```diff
@@ -1273,15 +1273,42 @@ def run(self, tau_max=20, start=0, stop=None, step=1, verbose=False):
         if tau_max > (stop - start):
             raise ValueError("Too few frames selected for given tau_max.")
 
-        # load all frames to an array of sets
+        # preload the frames (selected ids) to a list of sets
         selected_ids = []
-        for ts in self.universe.trajectory[start:stop]:
-            self.print(verbose, "Loading frame:", ts)
-            selected_ids.append(set(self.universe.select_atoms(self.selection).ids))
+        if step <= tau_max + 1:
+            # load all frames
+            for ts in self.universe.trajectory[start:stop]:
+                self.print(verbose, "Loading frame:", ts)
+                selected_ids.append(set(self.universe.select_atoms(self.selection).ids))
+        else:
+            # skip frames that will not be used
+            # Example: step 5 and tau 2: L, L, L, S, S, L, L, L, S, S, ... with L = Load, and S = Skip
+            loaded_counter = 0
```

Comment:
Why would you like to do this? Maybe I'm missing something, but when you calculate the Survival Probability, the definition says that you should consider only the particles that REMAIN in the selection between time t and t+tau, ie, you should load all the frames between t and t+tau and count in t, then count in t+1, count in t+2..., count in t+tau, counting only the particles that were in the selection in the previous step, so I don't understand why would you like to skip some frames when step is larger than the tau_max + 1. Maybe an other example could help to clarify this.

Reply:
Thanks @alejob 

It is always the case that we load all the frames between t and t+tau. The step foremost allows to remove the overlap of the measurement. I.e. a step larger than tau_max may be required to ensure that subsequent t0s are not correlated. To rephrase, if an atom survives from t to t+tau, then for the next 20 frames, the data for tau=1 will have 20 data points where the atoms survives.

Then there is a performance regarding sampling. If someone would like to scan through a long simulation, skipping frames is a huge performance gain. This data can be also used for discussion for equilibration and stability of the system.

Mat and @p-j-smith

Reply:
Great, now I understand, and this look pretty good. Thanks for your improvement.